driver/razermouse_driver.c

```diff
@@ -2655,6 +2765,13 @@ static int razer_mouse_probe(struct hid_device *hdev, const struct hid_device_id
         CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_device_mode);
 
         switch(dev->usb_pid) {
+        case USB_DEVICE_ID_RAZER_LANCEHEAD_WIRELESS:
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_charge_effect);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_charge_colour);
+        case USB_DEVICE_ID_RAZER_LANCEHEAD_WIRED:
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_charge_level);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_charge_status);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_charge_low_threshold);
```

Comment:
please add fallthrough comments there so the compiler is definitely happy? With the latest kernel version or so, these warnings are enabled by default in the kernel compilation.

Reply:
So when trying [[fallthrough]]; the compiler wasn't having it at all.  I found an article on GCC 7 and various fall-through statements.  The following compiled for me, and seemed like it should work with older compilers:

`__attribute__ ((fallthrough)); // C and C++03`

Do you think this is sufficient, or do you prefer another method?

For reference:  https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/

Reply:
Travis does not seem to like it all that much...

Reply:
Went with a comment vs an attribute.